File: fix_timestamps.py
import sqlite3
import time
from datetime import datetime

# Connect to database
conn = sqlite3.connect('db.sqlite')
cursor = conn.cursor()

# Get all users and their timestamps
cursor.execute('SELECT id, username, created_at FROM users')
users = cursor.fetchall()

print("Current user timestamps:")
print("-" * 60)
for user_id, username, created_at in users:
    date_str = datetime.fromtimestamp(created_at).strftime('%Y-%m-%d %H:%M:%S') if created_at > 0 else 'Invalid (0 or negative)'
    print(f"ID: {user_id}, Username: {username}, Timestamp: {created_at}, Date: {date_str}")

# Check for invalid timestamps (0 or very small values)
invalid_users = [u for u in users if u[2] < 1000000000]  # Before year 2001

if invalid_users:
    print(f"\n{len(invalid_users)} users with invalid timestamps found!")
    print("Updating to current time...")
    
    # Get current Unix timestamp
    current_timestamp = int(time.time())
    print(f"Current timestamp: {current_timestamp} ({datetime.fromtimestamp(current_timestamp)})")
    
    # Update all invalid timestamps
    for user_id, username, old_ts in invalid_users:
        cursor.execute('UPDATE users SET created_at = ? WHERE id = ?', (current_timestamp, user_id))
        print(f"Updated user {username} (ID: {user_id}) from {old_ts} to {current_timestamp}")
    
    conn.commit()
    print(f"\n✅ Updated {len(invalid_users)} user(s) successfully!")
else:
    print("\n✅ All user timestamps are valid!")

conn.close()

File: server/fix-timestamps.ts
// Temporary script to fix user timestamps
// Add this to routes.ts temporarily, then call via browser/curl

import { storage } from './storage';

export async function fixUserTimestamps() {
    const users = await storage.getUsers();
    const currentTimestamp = Math.floor(Date.now() / 1000); // Current Unix timestamp in seconds

    console.log('Checking user timestamps...');
    console.log('Current timestamp:', currentTimestamp, new Date(currentTimestamp * 1000).toISOString());

    const invalidUsers = [];

    for (const user of users) {
        const createdAtTimestamp = Math.floor(new Date(user.createdAt).getTime() / 1000);

        // Check if timestamp is before year 2020 (likely invalid)
        if (createdAtTimestamp < 1577836800) { // Jan 1, 2020
            console.log(`Invalid timestamp for user ${user.username} (ID: ${user.id}): ${createdAtTimestamp} (${new Date(createdAtTimestamp * 1000).toISOString()})`);
            invalidUsers.push(user);
        }
    }

    if (invalidUsers.length > 0) {
        console.log(`\nFound ${invalidUsers.length} users with invalid timestamps. Updating...`);

        // Import getSqlite
        const { getSqlite } = await import('./db');
        const sqlite = getSqlite();

        if (sqlite && process.env.USE_SQLITE === 'true') {
            const stmt = sqlite.prepare('UPDATE users SET created_at = ? WHERE id = ?');

            for (const user of invalidUsers) {
                stmt.run(currentTimestamp, user.id);
                console.log(`✅ Updated user ${user.username} (ID: ${user.id}) to ${new Date(currentTimestamp * 1000).toISOString()}`);
            }

            console.log(`\n✅ Successfully updated ${invalidUsers.length} user(s)!`);
            return { success: true, updated: invalidUsers.length };
        } else {
            console.log('❌ SQLite not available or not in use');
            return { success: false, message: 'SQLite not available' };
        }
    } else {
        console.log('✅ All user timestamps are valid!');
        return { success: true, updated: 0 };
    }
}
